### DISCUSSION TITLE
[BUG] 1.11.0 restore-pnprecyclebinItem fails

### DISCUSSION THREAD

**Discussion by PatrizioCG** [pnp/powershell#2411] [General]
Version 1.11.0 in Windows Enviroment

$DeletedItems | ? {$_.leafname -eq "xxxxxxConso.pdf"} | Restore-PnPRecycleBinItem -Force     

Restore-PnPRecycleBinItem : 

Value does not fall within the expected range.
In riga:1 car:76
+ ... nith xxxxxConso.pdf"} | Restore-PnPRecycleBinItem -Force
+                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Restore-PnPRecycleBinItem], ServerException
    + FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.RecycleBin.RestoreRecycleBinItem
Restore-PnPRecycleBinItem : Value does not fall within the expected range.
In riga:1 car:76
+ ... xxxxxConso.pdf"} | Restore-PnPRecycleBinItem -Force
+                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Restore-PnPRecycleBinItem], ServerException
    + FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.RecycleBin.RestoreRecycleBinItem

**Comment by gautamdsheth** [comment]
Moved this to discussion , looks like some issue with your script. 

Can you share more details about your script ?

**Comment by PatrizioCG** [comment]
hi, sorry, i try to be clear.
I've lunched that command first time and got this error:

```
$DeletedItems = Get-PnPRecycleBinItem -RowLimit 500000 | Where { $_.DeletedDate -gt (Get-Date).AddDays(-7) }
$DeletedItems | ? {$_.leafname -eq "Zenith Service AR 2015 - Conso.pdf"}

Title                              Id                                   ItemType LeafName                           Dir
                                                                                                                    Nam
                                                                                                                    e
-----                              --                                   -------- --------                           ---
Zenith Service AR 2015 - Conso.pdf 3a88cadf-fca7-4870-8c2b-24757cfcf60c File     Zenith Service AR 2015 - Conso.pdf sit
Zenith Service AR 2015 - Conso.pdf 9b5e542f-1401-4d0b-b579-c8b0cb247450 File     Zenith Service AR 2015 - Conso.pdf sit


$DeletedItems | ? {$_.leafname -eq "Zenith Service AR 2015 - Conso.pdf"} | Restore-PnPRecycleBinItem -Force

**Restore-PnPRecycleBinItem : The request message is too big. The server does not allow messages larger than 2097152**
bytes.
In riga:1 car:76
+ ... nith Service AR 2015 - Conso.pdf"} | Restore-PnPRecycleBinItem -Force
+                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Restore-PnPRecycleBinItem], ServerException
    + FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.RecycleBin.RestoreRecycleBinItem
```


Then, i've retried the command and got the error , but i've seend that the file was restored ...
